tests/perf-test/ReportUtils.perf-test.ts

```diff
@@ -0,0 +1,267 @@
+
+    const reportAction = {...createRandomReportAction(1), actionName: CONST.REPORT.ACTIONS.TYPE.ADDCOMMENT} as unknown as ReportAction;
+
+    await Onyx.multiSet({
+        ...mockedPoliciesMap,
+        ...mockedReportsMap,
+    });
+
+    await waitForBatchedUpdates();
+    await measureFunction(() => ReportUtils.canDeleteReportAction(reportAction, reportID), {runs});
+});
+
+test('[ReportUtils] getReportRecipientAccountID on 1k participants', async () => {
+    const report = {...createRandomReport(1), participantAccountIDs};
+    const currentLoginAccountID = 1;
+
+    await Onyx.multiSet({
+        ...mockedReportsMap,
+    });
+
+    await waitForBatchedUpdates();
+    await measureFunction(() => ReportUtils.getReportRecipientAccountIDs(report, currentLoginAccountID), {runs});
+});
+
+test('[ReportUtils] getIconsForParticipants on 1k participants', async () => {
+    const participants = Array.from({length: 1000}, (v, i) => i + 1);
+
+    await waitForBatchedUpdates();
+    await measureFunction(() => ReportUtils.getIconsForParticipants(participants, personalDetails), {runs});
+});
+
+test('[ReportUtils] getIcons on 1k participants', async () => {
+    const report = {...createRandomReport(1), parentReportID: '1', parentReportActionID: '1', type: CONST.REPORT.TYPE.CHAT};
+    const policy = createRandomPolicy(1);
+    const defaultIcon = null;
+    const defaultName = '';
+    const defaultIconId = -1;
+
+    await Onyx.multiSet({
+        [ONYXKEYS.PERSONAL_DETAILS_LIST]: personalDetails,
+    });
+
+    await waitForBatchedUpdates();
+    await measureFunction(() => ReportUtils.getIcons(report, personalDetails, defaultIcon, defaultName, defaultIconId, policy), {runs});
+});
+
+test('[ReportUtils] getDisplayNamesWithTooltips 1k participants', async () => {
+    const isMultipleParticipantReport = true;
+    const shouldFallbackToHidden = true;
+
+    await waitForBatchedUpdates();
+    await measureFunction(() => ReportUtils.getDisplayNamesWithTooltips(personalDetails, isMultipleParticipantReport, shouldFallbackToHidden), {runs});
+});
+
+test('[ReportUtils] getReportPreviewMessage on 5k policies', async () => {
+    const reportAction = createRandomReportAction(1);
+    const report = createRandomReport(1);
+    const policy = createRandomPolicy(1);
+    const shouldConsiderReceiptBeingScanned = true;
+    const isPreviewMessageForParentChatReport = true;
+
+    await Onyx.multiSet({
+        ...mockedPoliciesMap,
+    });
+
+    await waitForBatchedUpdates();
+    await measureFunction(() => ReportUtils.getReportPreviewMessage(report, reportAction, shouldConsiderReceiptBeingScanned, isPreviewMessageForParentChatReport, policy), {runs});
+});
+
+test('[ReportUtils] getModifiedExpenseMessage on 5k reports and policies', async () => {
+    const reportAction = {
+        ...createRandomReportAction(1),
+        actionName: CONST.REPORT.ACTIONS.TYPE.MODIFIEDEXPENSE,
+        originalMessage: {
+            originalMessage: {
```

Comment:
Here's an extra node level (the `originalMessage ` key is duplicated) that differs from how it looks in production. I updated this test in [this PR](https://github.com/Expensify/App/pull/30603), but now the performance test fails.

Reply:
Good catch

Reply:
@cristipaval The reason your [PR](https://github.com/Expensify/App/pull/30603) failed the performance tests is because the number of re-renders was increased due to the changes you made. Do you have the same error locally? I would try to see if there is a way to improve the performance of this new `getForReportAction()` method.

Reply:
could you please check [this](https://expensify.slack.com/archives/C01GTK53T8Q/p1702508204991019) and help me run the tests locally?